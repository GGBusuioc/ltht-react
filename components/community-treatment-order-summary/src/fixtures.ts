import {
  LypftCommunityTreatmentOrder,
  CarePlanStatusCode,
  PartialDateTimeKindCode,
  CarePlanIntentCode,
  Metadata,
} from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
}

const CommunityTreatmentOrderOne: LypftCommunityTreatmentOrder = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|communityTreatmentOrder',
  metadata: mockMetadata,
  consentToTreat: 'Consent Not Given',
  legalStatus: {
    coding: [{ display: 'Approved', code: '3309007' }],
    text: 'Approved',
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  restrictions: 'No restrictions',
}

const CommunityTreatmentOrderTwo: LypftCommunityTreatmentOrder = {
  id: '690ca927-68e0-c511-b2c7-898056926ze4|communityTreatmentOrder',
  metadata: mockMetadata,
  consentToTreat: 'Consent Given',
  legalStatus: {
    coding: [{ display: 'Pending', code: '779007' }],
    text: 'Pending',
  },
  period: {
    start: {
      value: '2020-01-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  restrictions: 'No Restrictions',
}

export { CommunityTreatmentOrderOne, CommunityTreatmentOrderTwo }
