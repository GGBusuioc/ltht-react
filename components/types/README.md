# `ltht-types`

> TODO: description

## Usage

```
const lthtTypes = require('ltht-types');

// TODO: DEMONSTRATE API
```
