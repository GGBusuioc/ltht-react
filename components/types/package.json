{
  "name": "@ltht-react/types",
  "version": "0.0.1-alpha.1",
  "description": "> TODO: description",
  "author": "Jonny Smith <jonny.smith@digitaljs.com>",
  "homepage": "https://github.com/ltht-epr/ltht-react/tree/master/components/ltht-types#readme",
  "license": "MIT",
  "main": "lib/index.js",
  "typings": "lib/index.d.ts",
  "directories": {
    "lib": "lib",
    "test": "__tests__"
  },
  "files": [
    "lib"
  ],
  "publishConfig": {
    "registry": "https://registry.yarnpkg.com"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/ltht-epr/ltht-react.git"
  },
  "scripts": {
    "build": "yarn build:lib && yarn build:es && yarn build:definition",
    "build:lib": "rimraf lib && babel src -d lib --extensions '.ts,.tsx' --source-maps --config-file ../../babel.config.js",
    "build:es": "rimraf es && cross-env BABEL_ENV=es babel src -d es --extensions '.ts,.tsx' --source-maps --config-file ../../babel.config.js",
    "build:definition": "tsc --noEmit false --emitDeclarationOnly",
    "build:watch": "tsc --noEmit false --emitDeclarationOnly --watch",
    "type:check": "tsc"
  },
  "bugs": {
    "url": "https://github.com/ltht-epr/ltht-react/issues"
  },
  "gitHead": "4b363f052431bd35738693f2a5986caa24c841cf"
}
