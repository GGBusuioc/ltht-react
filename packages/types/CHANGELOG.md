# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.0.2](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/types@0.0.1...@ltht-react/types@0.0.2) (2020-01-06)

**Note:** Version bump only for package @ltht-react/types





## [0.0.1](https://github.com/ltht-epr/ltht-react/compare/@ltht-react/types@0.0.1-alpha.3...@ltht-react/types@0.0.1) (2020-01-03)

**Note:** Version bump only for package @ltht-react/types
