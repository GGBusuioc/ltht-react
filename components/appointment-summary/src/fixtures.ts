import { Encounter, EncounterStatusCode, PartialDateTimeKindCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
}

const AppointmentOne: Encounter = {
  id: '346ca927-67e0-e931-a2c7-115087226fg2|appointment',
  metadata: mockMetadata,
  type: [
    {
      text: 'See GP regarding sore throat',
      coding: [{ display: 'GP', code: '64706' }],
    },
  ],
  period: {
    start: {
      value: '2020-04-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  serviceProvider: {
    display: 'Beeston Village Surgery',
    typeName: 'GP',
  },
  status: EncounterStatusCode.Planned,
}

const AppointmentTwo: Encounter = {
  id: '941ca927-34e0-e911-a2c7-005087226fe4|appointment',
  metadata: mockMetadata,
  type: [
    {
      text: 'Chest x-ray for recent car accident',
      coding: [{ display: 'Hospital', code: '26706' }],
    },
  ],
  period: {
    start: {
      value: '2019-12-12T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  serviceProvider: {
    display: 'Chapel Allerton Hospital',
    typeName: 'Hospital',
  },
  status: EncounterStatusCode.Finished,
}

export { AppointmentOne, AppointmentTwo }
