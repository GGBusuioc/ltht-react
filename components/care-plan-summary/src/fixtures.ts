import { CarePlan, CarePlanStatusCode, PartialDateTimeKindCode, CarePlanIntentCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
}

const CarePlanOne: CarePlan = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|careplan',
  metadata: mockMetadata,
  author: { display: 'Joe Bloggs', type: 'Test' },
  intent: CarePlanIntentCode.Plan,
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: CarePlanStatusCode.Active,
  title: 'Test Care Plan Summary One',
}

const CarePlanTwo: CarePlan = {
  id: '690ca927-68e0-c511-b2c7-898056926ze4|careplan',
  metadata: mockMetadata,
  author: { display: 'John Mclane', type: 'Test' },
  intent: CarePlanIntentCode.Proposal,
  period: {
    start: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2015-03-04T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
  status: CarePlanStatusCode.Completed,
  title: 'Test Care Plan Summary Two',
}

export { CarePlanOne, CarePlanTwo }
