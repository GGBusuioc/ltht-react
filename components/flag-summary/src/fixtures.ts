import { Flag, FlagStatusCode, PartialDateTimeKindCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
}

const FlagOne: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    type: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'Lives in Known High Risk Area', code: '109006' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Year,
    },
  },
  status: FlagStatusCode.Inactive,
}

const FlagTwo: Flag = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|personalertandhazard',
  metadata: mockMetadata,
  author: {
    display: 'John Smith',
    reference:
      'https://phoenixqa.careworks.ie/Services/api/fhir/v3/practitioner/9c1ad2c0-ec34-e911-a2c5-005056926fe4|systemuser',
    type: 'Practitioner',
    identifier: undefined,
  },
  code: {
    coding: [{ display: 'High Risk of Violence', code: '56706' }],
  },
  category: {
    coding: [{ display: 'Violence', code: '66706' }],
  },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },

  status: FlagStatusCode.Inactive,
}

export { FlagOne, FlagTwo }
