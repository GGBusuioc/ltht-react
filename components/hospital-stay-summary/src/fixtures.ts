import { HospitalStay, PartialDateTimeKindCode, Metadata } from '@ltht-react/types'

const mockMetadata: Metadata = {
  dataSources: [
    {
      display: 'Mock',
    },
  ],
}

const HospitalStayOne: HospitalStay = {
  id: '580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay',
  metadata: mockMetadata,
  nhsNumber: '123456789',
  serviceProvider: { display: 'Chapel Allerton Hospital', type: 'Hospital' },
  period: {
    start: {
      value: '2013-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
    end: {
      value: '2014-02-02T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
}

const HospitalStayTwo: HospitalStay = {
  id: '690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay',
  metadata: mockMetadata,
  nhsNumber: '987654321',
  serviceProvider: { display: 'Beaumont Hospital', type: 'Hospital' },
  period: {
    start: {
      value: '2020-01-01T00:00:00+00:00',
      kind: PartialDateTimeKindCode.Date,
    },
  },
}

export { HospitalStayOne, HospitalStayTwo }
