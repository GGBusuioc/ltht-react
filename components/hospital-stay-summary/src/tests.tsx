import React from 'react'
import { mount } from 'enzyme'
import ReactDOM from 'react-dom'

import { HospitalStay } from '@ltht-react/types'
import { HospitalStayOne, HospitalStayTwo } from './fixtures'
import HospitalStaySummary from '.'

const handleClick = (hospitalStay: HospitalStay) => {}

describe('HospitalStaySummary', () => {
  const hospitalStays = [HospitalStayOne, HospitalStayTwo]
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <HospitalStaySummary title="Hospital Stays" hospitalStays={hospitalStays} clickHandler={handleClick} />,
      div
    )
  })
  it('matches snapshot', () => {
    expect(
      mount(<HospitalStaySummary title="Hospital Stays" hospitalStays={hospitalStays} clickHandler={handleClick} />)
    ).toMatchSnapshot('wrapper mount')
  })
})
