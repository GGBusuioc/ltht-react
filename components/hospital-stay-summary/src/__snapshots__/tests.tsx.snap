// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HospitalStaySummary matches snapshot: wrapper mount 1`] = `
.emotion-12 {
  color: #231f20;
  font-family: 'SF Pro Text','SF Pro Icons','Helvetica Neue',Helvetica,Arial,sans-serif;
  font-size: 0.8rem;
  background: #fff;
  color: #000;
  margin-bottom: 0.5rem;
  border-radius: 5px;
  box-shadow: 3px 3px 3px #666;
  -webkit-box-shadow: 3px 3px 3px #666;
  margin: 0 5px 10px 0;
  -webkit-font-smoothing: antialiased;
}

.emotion-12 *,
.emotion-12 ::before,
.emotion-12 ::after {
  box-sizing: border-box;
  padding: 0;
  margin: 0;
}

.emotion-12 * {
  margin: 0;
  padding: 0;
}

.emotion-12 h3 {
  font-size: 1rem;
}

@media only screen and (min-width:768px) {
  .emotion-12 {
    font-size: 0.9rem;
  }

  .emotion-12 h3 {
    font-size: 1.1rem;
  }
}

@media only screen and (min-width:1024px) {
  .emotion-12 {
    font-size: 1rem;
  }

  .emotion-12 h3 {
    font-size: 1.2rem;
  }
}

@media only screen and (min-width:1200px) {
  .emotion-12 {
    font-size: 1.1rem;
  }

  .emotion-12 h3 {
    font-size: 1.3rem;
  }
}

.emotion-0 {
  padding: 0.5rem;
}

.emotion-11 {
  list-style: none;
  padding-left: 0.5rem;
}

.emotion-11 li {
  border-top: 1px solid #b0b0b0;
  padding: 0.5rem 0.5rem 0.5rem 0;
}

.emotion-11 li:hover {
  cursor: pointer;
  background: #E8EDEE;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-1 {
  color: #231f20;
  text-align: left;
}

.emotion-3 {
  color: #768692;
  text-align: right;
}

<HospitalStaySummary
  clickHandler={[Function]}
  hospitalStays={
    Array [
      Object {
        "id": "580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
        },
        "period": Object {
          "end": Object {
            "kind": "DATE",
            "value": "2014-02-02T00:00:00+00:00",
          },
          "start": Object {
            "kind": "DATE",
            "value": "2013-01-01T00:00:00+00:00",
          },
        },
        "serviceProvider": Object {
          "display": "Chapel Allerton Hospital",
          "typeName": "Hospital",
        },
      },
      Object {
        "id": "690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay",
        "metadata": Object {
          "dataSources": Array [
            Object {
              "display": "Mock",
            },
          ],
        },
        "period": Object {
          "start": Object {
            "kind": "DATE",
            "value": "2020-01-01T00:00:00+00:00",
          },
        },
        "serviceProvider": Object {
          "display": "Beaumont Hospital",
          "typeName": "Hospital",
        },
      },
    ]
  }
  title="Hospital Stays"
>
  <Widget>
    <div
      className="emotion-12"
    >
      <WidgetHeader>
        <div
          className="emotion-0"
        >
          <h3>
            Hospital Stays
          </h3>
        </div>
      </WidgetHeader>
      <WidgetList
        clickable={true}
      >
        <ul
          className="emotion-11"
        >
          <WidgetListItem
            key="0"
          >
            <li>
              <HospitalStaySummaryItem
                clickHandler={[Function]}
                hospitalStay={
                  Object {
                    "id": "580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                    },
                    "period": Object {
                      "end": Object {
                        "kind": "DATE",
                        "value": "2014-02-02T00:00:00+00:00",
                      },
                      "start": Object {
                        "kind": "DATE",
                        "value": "2013-01-01T00:00:00+00:00",
                      },
                    },
                    "serviceProvider": Object {
                      "display": "Chapel Allerton Hospital",
                      "typeName": "Hospital",
                    },
                  }
                }
              >
                <div
                  className="emotion-5"
                  onClick={[Function]}
                >
                  <div
                    className="emotion-2"
                  >
                    <HospitalStayPeriod
                      hospitalStay={
                        Object {
                          "id": "580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay",
                          "metadata": Object {
                            "dataSources": Array [
                              Object {
                                "display": "Mock",
                              },
                            ],
                          },
                          "period": Object {
                            "end": Object {
                              "kind": "DATE",
                              "value": "2014-02-02T00:00:00+00:00",
                            },
                            "start": Object {
                              "kind": "DATE",
                              "value": "2013-01-01T00:00:00+00:00",
                            },
                          },
                          "serviceProvider": Object {
                            "display": "Chapel Allerton Hospital",
                            "typeName": "Hospital",
                          },
                        }
                      }
                    >
                      <div
                        className="emotion-1"
                      >
                        01-Jan-2013 to 02-Feb-2014
                      </div>
                    </HospitalStayPeriod>
                  </div>
                  <div
                    className="emotion-2"
                  >
                    <HospitalStayServiceProvider
                      hospitalStay={
                        Object {
                          "id": "580ca927-34e0-e911-a2c7-005056926fe4|hospitalstay",
                          "metadata": Object {
                            "dataSources": Array [
                              Object {
                                "display": "Mock",
                              },
                            ],
                          },
                          "period": Object {
                            "end": Object {
                              "kind": "DATE",
                              "value": "2014-02-02T00:00:00+00:00",
                            },
                            "start": Object {
                              "kind": "DATE",
                              "value": "2013-01-01T00:00:00+00:00",
                            },
                          },
                          "serviceProvider": Object {
                            "display": "Chapel Allerton Hospital",
                            "typeName": "Hospital",
                          },
                        }
                      }
                    >
                      <div
                        className="emotion-3"
                      >
                        Chapel Allerton Hospital
                      </div>
                    </HospitalStayServiceProvider>
                  </div>
                </div>
              </HospitalStaySummaryItem>
            </li>
          </WidgetListItem>
          <WidgetListItem
            key="1"
          >
            <li>
              <HospitalStaySummaryItem
                clickHandler={[Function]}
                hospitalStay={
                  Object {
                    "id": "690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay",
                    "metadata": Object {
                      "dataSources": Array [
                        Object {
                          "display": "Mock",
                        },
                      ],
                    },
                    "period": Object {
                      "start": Object {
                        "kind": "DATE",
                        "value": "2020-01-01T00:00:00+00:00",
                      },
                    },
                    "serviceProvider": Object {
                      "display": "Beaumont Hospital",
                      "typeName": "Hospital",
                    },
                  }
                }
              >
                <div
                  className="emotion-5"
                  onClick={[Function]}
                >
                  <div
                    className="emotion-2"
                  >
                    <HospitalStayPeriod
                      hospitalStay={
                        Object {
                          "id": "690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay",
                          "metadata": Object {
                            "dataSources": Array [
                              Object {
                                "display": "Mock",
                              },
                            ],
                          },
                          "period": Object {
                            "start": Object {
                              "kind": "DATE",
                              "value": "2020-01-01T00:00:00+00:00",
                            },
                          },
                          "serviceProvider": Object {
                            "display": "Beaumont Hospital",
                            "typeName": "Hospital",
                          },
                        }
                      }
                    >
                      <div
                        className="emotion-1"
                      >
                        01-Jan-2020 to Present
                      </div>
                    </HospitalStayPeriod>
                  </div>
                  <div
                    className="emotion-2"
                  >
                    <HospitalStayServiceProvider
                      hospitalStay={
                        Object {
                          "id": "690ca927-68e0-c511-b2c7-898056926ze4|hospitalstay",
                          "metadata": Object {
                            "dataSources": Array [
                              Object {
                                "display": "Mock",
                              },
                            ],
                          },
                          "period": Object {
                            "start": Object {
                              "kind": "DATE",
                              "value": "2020-01-01T00:00:00+00:00",
                            },
                          },
                          "serviceProvider": Object {
                            "display": "Beaumont Hospital",
                            "typeName": "Hospital",
                          },
                        }
                      }
                    >
                      <div
                        className="emotion-3"
                      >
                        Beaumont Hospital
                      </div>
                    </HospitalStayServiceProvider>
                  </div>
                </div>
              </HospitalStaySummaryItem>
            </li>
          </WidgetListItem>
        </ul>
      </WidgetList>
    </div>
  </Widget>
</HospitalStaySummary>
`;
